### PR TITLE
Add Community Settings Editor

### LINKED ISSUE #3493: community settings editor
refs. https://github.com/steemit/condenser/issues/2411

 - `title`: the display name of this community (32 chars)
 - `about`: short blurb about this community (120 chars)
 - `is_nsfw`: true if this community is 18+ (checkbox)
 - `description`: describes purpose of community, etc. (5000 chars)
 - `flag_text`: custom text for reporting content (2000 chars)

https://github.com/steemit/hivemind/blob/214-core-notifs2/docs/communities.md#editable-by-admins---community-properties

Op:
```
["updateProps", {
  "community": <community>, 
  "props": { <key:value>, ... }
}]
```

### PR DESCRIPTION
**Pull request by JonathanPorta**
# Screenshots:
image (15)
image (16)
image (17)

# Issues:
- [ ] Need to use a saga in order to promote refreshing the community page when settings change.
- [ ] Settings don't appear to have changed sometimes despite a successful broadcast transaction: [worked](https://steemd.com/b/36579958#a259d17ca3d8fed6a85681e2de373a98f1dc5af4) and [didn't work](https://steemd.com/b/36580938#16ba8aa0090a3dbc6920597153fc01595c00afd4)
- [ ] UI/layout needs love and attention.
- [ ] i18n remaining copy.

Fixes #3493

### CODE REVIEW

**Comment by roadscape** [comment]
The ops which didn't work were due to `key 'flag_text' was blank`. We could either make this field optional (i.e. do not add it to payload if its blank), or clarify that it's required.

**Comment by roadscape** [comment]
Found another issue, all input needs to be trimmed of whitespace before crafting payload

**Comment by roadscape** [comment]
Remaining issues:
 - Need to use a saga in order to promote refreshing the community page when settings change
 - all input needs whitespace trim before crafting payload
 - `flag_text` should be excluded from the op if blank

File: src/app/components/modules/CommunitySettings.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    onSubmit: PropTypes.func.isRequired,
+    title: PropTypes.string.isRequired,
+    about: PropTypes.string.isRequired,
+    is_nsfw: PropTypes.bool.isRequired,
+    description: PropTypes.string.isRequired,
+    flag_text: PropTypes.string.isRequired,
+};
+
+CommunitySettings.defaultProps = {
+    isMuted: false,
```
Comment (JonathanPorta) [inline src/app/components/modules/CommunitySettings.jsx]: :roll_eyes:

**Pull request merged**